stdlib/public/stubs/LibcShims.cpp

```diff
@@ -329,3 +332,60 @@ swift::_stdlib_cxx11_mt19937_uniform(__swift_uint32_t upper_bound) {
   std::uniform_int_distribution<__swift_uint32_t> RandomUniform(0, upper_bound);
   return RandomUniform(getGlobalMT19937());
 }
+
+#if defined(__APPLE__)
+#include "TargetConditionals.h"
+#if defined(TARGET_IPHONE_SIMULATOR) \
+  || ( defined(TARGET_OS_IPHONE) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 10000 ) \
+  || ( defined(TARGET_OS_MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 101200 )
+SWIFT_RUNTIME_STDLIB_INTERFACE
+void swift::_stdlib_random(void *buf,
+                                __swift_ssize_t nbytes,
+                                __swift_uint32_t debug_flags) {
```

Comment:
* Should `arc4random_buf` vs `SecRandomCopyBytes` use [__builtin_available](https://clang.llvm.org/docs/LanguageExtensions.html#objective-c-available)?

* Should `__swift_ssize_t` be `__swift_size_t` instead?

* Can the `debug_flags` parameter be removed?
  Other APIs always call `fatalError` with zero flags.

* Reopening "/dev/urandom" for each `_stdlib_random` call seems inefficient.

```cpp
#if defined(__APPLE__)
#include <Security/Security.h>
SWIFT_RUNTIME_STDLIB_INTERFACE
void swift::_stdlib_random(void *buf, __swift_size_t nbytes) {
  if (__builtin_available(macOS 10.12, iOS 10, tvOS 10, watchOS 3, *)) {
    arc4random_buf(buf, nbytes);
  } else {
    OSStatus status {SecRandomCopyBytes(kSecRandomDefault, nbytes, buf)};
    if (status != errSecSuccess) {
      fatalError(0, "SecRandomCopyBytes failed with error %d\n", status);
    }
  }
}
#endif
```

Reply:
Thanks!